aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -4,6 +4,24 @@
 
 namespace c10 {
 
+class OpKernel final {
+public:
+  explicit constexpr OpKernel(KernelFunction* kernel): kernel_(kernel) {}
+
+  OpKernel(OpKernel&&) = default;
+  OpKernel& operator=(OpKernel&&) = default;
+  OpKernel(const OpKernel&) = delete;
+  OpKernel& operator=(const OpKernel&) = delete;
+
+  IValue call(ArrayRef<IValue> args) const {
+    return (*kernel_)(args);
+  }
+
+private:
+  // TODO Store kernel state
+  KernelFunction* kernel_;
```

Comment:
Would be nice to comment here that `KernelFunction` is expected to have static lifetime so ownership is not an issue here.